client/base/src/main/java/io/a2a/A2A.java

```diff
@@ -60,6 +63,108 @@ public static Message toAgentMessage(String text, String messageId) {
         return toMessage(text, Message.Role.AGENT, messageId);
     }
 
+    /**
+     * Create a user message with additional configuration options.
+     *
+     * @param text the message text
+     * @param contextId the context ID to use (optional)
+     * @param taskId the task ID to use (optional)
+     * @param parts the message parts (optional, defaults to a TextPart with the given text)
+     * @param metadata the message metadata (optional)
+     * @param referenceTaskIds the reference task IDs (optional)
+     * @return the user message
+     */
+    public static Message createUserMessage(String text, String contextId, String taskId, 
+                                           List<Part<?>> parts, Map<String, Object> metadata,
+                                           List<String> referenceTaskIds) {
+        return createMessage(text, Message.Role.USER, null, contextId, taskId, parts, metadata, referenceTaskIds);
+    }
+
+    /**
+     * Create a user message with additional configuration options.
+     *
+     * @param text the message text
+     * @param messageId the message ID to use
+     * @param contextId the context ID to use (optional)
+     * @param taskId the task ID to use (optional)
+     * @param parts the message parts (optional, defaults to a TextPart with the given text)
+     * @param metadata the message metadata (optional)
+     * @param referenceTaskIds the reference task IDs (optional)
+     * @return the user message
+     */
+    public static Message createUserMessage(String text, String messageId, String contextId, String taskId, 
+                                           List<Part<?>> parts, Map<String, Object> metadata,
+                                           List<String> referenceTaskIds) {
+        return createMessage(text, Message.Role.USER, messageId, contextId, taskId, parts, metadata, referenceTaskIds);
+    }
+
+    /**
+     * Create an agent message with additional configuration options.
+     *
+     * @param text the message text
+     * @param contextId the context ID to use (optional)
+     * @param taskId the task ID to use (optional)
+     * @param parts the message parts (optional, defaults to a TextPart with the given text)
+     * @param metadata the message metadata (optional)
+     * @param referenceTaskIds the reference task IDs (optional)
+     * @return the agent message
+     */
+    public static Message createAgentMessage(String text, String contextId, String taskId, 
```

Comment:
Similarly, maybe here instead of this method signature, we could use something like `public static Message createAgentTextMessage(String text, String contextId, String taskId)` where `contextId` and `taskId` are optional to simplify the number the of method parameters here.

This method could then call the private `toMessage` method which can be updated to add the additional parameters.

We could also add a `createAgentPartsMessage` method that takes `List<Part<?>> parts` (required) and `contextId` (optional) and `taskId` (optional).